packages/compiler/src/yaml/types.ts

```diff
@@ -0,0 +1,26 @@
+import { Document } from "yaml";
+import { SourceFile } from "../core/types.js";
+
+export interface YamlScript {
+  readonly kind: "yaml-script";
+  readonly file: SourceFile;
+  /** Value of the yaml script. */
+  readonly value: unknown;
+
+  /** @internal yaml library document. We do not expose this as the "yaml" library is not part of the contract. */
+  readonly doc: Document.Parsed;
+}
+
+/**
+ * Diagnostic target pointing to a specific yaml node.
+ */
+export interface YamlDiagnosticTarget {
+  /** Yaml script */
+  readonly file: YamlScript;
+  /** Path to the target node from the root of the document. */
+  readonly path: string[];
+  /** If targeting the value or the key in the case of a map. */
+  readonly kind: YamlDiagnosticTargetType;
+}
```

Comment:
Does this type seem valuable to people or would it just be bettrer to have 
```
getYamlScriptLocation(script, ["path", "to", "node"]): SourceLocation
```
and you call that for every diagnostic you want to emit

Reply:
Feel like it is the same amount of effort when reporting but add extra conversion layer